src/OrleansRuntime/Startup/SiloHostBuilder.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.DependencyInjection;
+using Orleans.Runtime.Configuration;
+using Orleans.Runtime.Host;
+
+namespace Orleans.Runtime.Startup
+{
+    public class SiloHostBuilder : ISiloHostBuilder
+    {
+        private ClusterConfiguration clusterConfiguration;
+        private readonly List<Action<IServiceCollection>> configureServicesDelegates = new List<Action<IServiceCollection>>();
```

Comment:
It isn't actually!

There could be tons of extension methods that want to configure the service collection.  If it is called more than once it would overwrite.  In the ASP.NET builder code (and my own applications) you need a list to organize different functionality with it's own initialization code (e.g. AddLogging(), AddTelemetry) which all need to add things to the `IServiceCollection`